Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -1130,17 +1127,22 @@ class TabCell: UICollectionViewCell {
         isAccessibilityElement = true
         accessibilityHint = NSLocalizedString("Swipe right or left with three fingers to close the tab.", comment: "Accessibility hint for tab tray's displayed tab.")
 
-        if let favIcon = tab.displayFavicon, let url = URL(string: favIcon.url) {
-            favicon.sd_setImage(with: url, placeholderImage: UIImage(named: "defaultFavicon"), options: [], completed: nil)
-        } else {
-            let defaultFavicon = UIImage(named: "defaultFavicon")
-            if tab.isPrivate {
-                favicon.image = defaultFavicon
-                favicon.tintColor = UIColor.theme.tabTray.faviconTint
-            } else {
-                favicon.image = defaultFavicon
-            }
+        let logoPlaceholderURL = URL(string: Strings.BrandWebsite)!
+
+        let faviconURL = URL(nullableString: tab.displayFavicon?.url)
+
+        var logoURL = tab.url ?? faviconURL ?? logoPlaceholderURL
+
+        if InternalURL.isValid(url: logoURL) {
+            logoURL = logoPlaceholderURL
         }
+
+        let logo = LogoView(
+            frame: CGRect(x: 0, y: 0, width: TabTrayControllerUX.FaviconSize, height: TabTrayControllerUX.FaviconSize),
+            url: logoURL.absoluteString
+        )
+        favicon.addSubview(logo)
+
```

Comment:
This code is repeated here, in `PhotonActionSheetSiteHeaderView`, and in `BackForwardTableViewCell.swift`. Once the same or almost the same code is repeated more than 2 times, that's the right time to refactor. 

I recommend putting it into a custom UIView subclass and using that instead. This way you can just init the custom view in TabCell:1026 and pass it its size if necessary.

Reply:
Created Ticket #246 to fix this later